src/ripple/rpc/impl/ServerHandlerImp.cpp

```diff
@@ -874,7 +874,17 @@ ServerHandlerImp::processRequest(
             {user, forwardedFor}};
         Json::Value result;
         auto start = std::chrono::system_clock::now();
-        RPC::doCommand(context, result);
+        try
+        {
+            RPC::doCommand(context, result);
+        }
+        catch (std::exception const& ex)
+        {
+            result[jss::result] = RPC::make_error(rpcINTERNAL);
+            JLOG(m_journal.error())
+                << "Exception while processing WS: " << ex.what() << "\n"
+                << "Input JSON: " << Json::Compact{Json::Value{result}};
```

Comment:
Are we sure that attempting to print the (potentially) offending JSON won't, itself, generate an exception?

Also, I think that `result[jss::result] = RPC::make_error(rpcINTERNAL);` may be insufficient in some cases. See the code around line 896.

Reply:
I think we want to print the request here, not `result`